### PR TITLE
Fix inline comment position in inline diff view.

### LINKED ISSUE #1099: Peek view sometimes shows up in wrong place in inline diff mode
For example this comment:

https://github.com/AvaloniaUI/Avalonia/pull/827#discussion_r126401935

Appears on the wrong line:

image

### PR DESCRIPTION
**Pull request by grokys**
We need to map between projection buffers: we were previously doing it for deleted lines, but we need to do it for all lines in the inline diff view.

Fixes #1099

### CODE REVIEW

**Comment by shana** [comment]
Currently getting this behaviour:

pr-1101

**Comment by rubenscholfield** [comment]
Canada in Come around course of work

**Comment by rubenscholfield** [comment]
Can in come around course of workk

**Comment by rubenscholfield** [comment]
???
need your adress

**Comment by rubenscholfield** [comment]
Email rubenscholfield@gmail.com

**Comment by rubenscholfield** [comment]
Greatings from germany

File: src/GitHub.InlineReviews/Services/InlineCommentPeekService.cs [hunk 1]
```diff
@@ -51,18 +51,18 @@ class InlineCommentPeekService : IInlineCommentPeekService
             {
                 if (diffModel.ViewType == DifferenceViewType.InlineView)
                 {
-                    // If we're displaying a diff in inline mode, then we're in the left buffer if
```
Comment (grokys) [inline src/GitHub.InlineReviews/Services/InlineCommentPeekService.cs]: // If we're displaying a diff in inline mode, then we're in the left buffer if

File: src/GitHub.InlineReviews/Services/InlineCommentPeekService.cs [hunk 2]
```diff
@@ -51,18 +51,18 @@ class InlineCommentPeekService : IInlineCommentPeekService
...
-                    // the point can be mapped down to the left buffer.
+                    // If we're displaying a diff in inline mode, then we need to map the point down
```
Comment (grokys) [inline src/GitHub.InlineReviews/Services/InlineCommentPeekService.cs]: // If we're displaying a diff in inline mode, then we need to map the point down

File: src/GitHub.InlineReviews/Services/InlineCommentPeekService.cs [hunk 3]
```diff
@@ -51,18 +51,18 @@ class InlineCommentPeekService : IInlineCommentPeekService
...
+                    // to the left or right buffer.
                     var snapshotPoint = point.GetPoint(point.TextBuffer.CurrentSnapshot);
-                    var mappedPoint = session.TextView.BufferGraph.MapDownToBuffer(
+                    var mappedPoint = session.TextView.BufferGraph.MapDownToFirstMatch(
                         snapshotPoint,
                         PointTrackingMode.Negative,
```
Comment (grokys) [inline src/GitHub.InlineReviews/Services/InlineCommentPeekService.cs]: PointTrackingMode.Negative,

**Comment by jcansdale** [comment]
Testing *without* this PR on the `PullRequestsSandbox`. Everything is fine up until line 14, when it appears on the line above where it's supposed to.

image

Testing with this PR, it appears to be fixed!

image

**Pull request merged**